include/mbedtls/x509_crt.h

```diff
@@ -175,19 +175,34 @@ int mbedtls_x509_crt_parse_der( mbedtls_x509_crt *chain, const unsigned char *bu
                         size_t buflen );
 
 /**
- * \brief          Parse one or more certificates and add them
- *                 to the chained list. Parses permissively. If some
- *                 certificates can be parsed, the result is the number
- *                 of failed certificates it encountered. If none complete
- *                 correctly, the first error is returned.
+ * \brief          Parse one DER-encoded or multiple concatenated PEM-encoded
+ *                 certificates and add them to the chained list.
+ *
+ *                 For PEM-encoded CRTs, the function parses permissively:
+ *                 If at least one certificate can be parsed, the function
+ *                 returns the number of certificates for which parsing failed
+ *                 (hence \c 0 if all certificates were parsed successfully).
+ *                 If no certificate could be parsed, the function returns
+ *                 the first (negative) error encountered during parsing.
+ *
+ *                 PEM encoded certificates may be interleaved by other data
+ *                 such as human readable descriptions of their content, as
+ *                 long as the certificates are enclosed in the PEM specific
+ *                 '-----{BEGIN/END} CERTIFICATE-----' delimiters.
+ *
+ * \param chain    The chain to which to add the parsed certificates.
+ * \param buf      The buffer holding the certificate data in PEM or DER format.
+ *                 For certificates in PEM encoding, this may be a concatenation
+ *                 of multiple certificates; for DER encoding, the buffer must
+ *                 comprise exactly one certificate.
+ * \param buflen   The size of \p buf, including the terminating \c NULL byte
+ *                 in case of PEM encoded data.
+ *
+ * \return         \c 0 if all certificates were parsed successfully.
+ * \return         The (positive) number of certificates that couldn't
+ *                 be parsed if parsing was partly successful (see above).
+ * \return         A negative X509- or PEM-specific error code otherwise.
```

Comment:
What is a negative X509?
In addition, typo: X509-

Our error codes are negative, so we could return the previous description:
"or a specific X509 or PEM error code"

Reply:
It is the abbreviation for X509-specific or PEM-specific.

Reply:
OK, but still, no point on specifying that these are negative, as the error codes are negative.
Personally, I prefer the previous phrase, as I mentioned.
I think we should consult a native English speaker though

Reply:
I mostly switched back to the old wording, but I kept `negative` because I think it's important to emphasize that here, in the presence of meaningful positive return values.

Reply:
@RonEld While our error codes are negative, the documentations usually state `non-zero error code`, suggesting `ret != 0` checks. So, from the documentation the user cannot infer that all our error codes are negative.